src/shared/install.c

```diff
@@ -743,8 +743,14 @@ static int remove_marked_symlinks_fd(
                                         continue;
                                 }
 
+                                r = path_extract_filename(dest, &fname);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to extract filename from path %s: %m", dest);
+                                        return r;
+                                }
```

Comment:
log_error_errno() and similar calls, return the error code you pass as first arg also back. (specifically, it returns `-abs(r)`)

This is useful, so that you can shorten the above to:

```
if (r < 0)
        return log_debug_errno(r, "…");
```